### ISSUE TITLE
Some lines in exported image are missing syntax highlighting

### ISSUE THREAD

**Issue by cyberpirate92** [carbon-app/carbon#1046]
**Describe the bug**
When trying to create an image for a javascript code snippet, some lines are missing syntax highlighting. 

**To Reproduce**

1. Go to https://carbon.now.sh/ooidaopSOldcdPg1HR5s
2. Click on Export button
3. Open the downloaded image and you can see that lines 3,5,11 and 13 don't have syntax highlighting although it's valid code.

**Expected behavior**
All lines in the exported image should have syntax highlighting. 

**Screenshots**
![image](https://user-images.githubusercontent.com/10222413/86183692-33429200-bb50-11ea-8d31-c88fccade3e1.png)

**Info (please complete the following information):**

- OS [e.g. macOS, Linux, Windows, iOS]: macOS
- Browser [e.g. Chrome, Safari, Firefox]: Chrome (Version 83), Safari (Version 13.1.1), Firefox (Version 77.0.1)

<details>
  <summary>
console.log('Regular');

console.log('%cColored text ', 'color: #f00;');

console.log('%cBold Text', 'font-weight: bold;');

console.log('%cItalic Text', 'font-style: italic;');

console.log('%cStrikethrough Text', 'text-decoration: line-through;');

console.log('%cColored text with background ', 'color: #f00; background-color: #000;');

console.log('%cColored text with background and custom font size ', 
'color: #f00; background-color: #000; font-size: 18px;');

// Insert '%c' at the position from which you want styling to begin
console.log('Styling can also be %cstarted from a specific part of the log', 
'font-size: 18px; color: green');
  </summary>
  <pre>
  https://carbon.now.sh/ooidaopSOldcdPg1HR5s
  </pre>
</details>

**Comment by cyberpirate92** [comment]
**Console log:**
```
Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/seti.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/codemirror.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/3024-night.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/blackboard.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/base16-dark.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/base16-light.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/cobalt.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/dracula.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/duotone-dark.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/hopscotch.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/lucario.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/material.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/monokai.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/oceanic-next.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/panda-syntax.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/paraiso-dark.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/seti.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
27c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/solarized.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/twilight.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/yeti.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/zenburn.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/seti.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/codemirror.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/3024-night.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/blackboard.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/base16-dark.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/base16-light.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/cobalt.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/dracula.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/duotone-dark.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/hopscotch.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/lucario.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/material.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/monokai.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/oceanic-next.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/panda-syntax.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/paraiso-dark.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/seti.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
27c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/solarized.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/twilight.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/yeti.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
7c6a7c0b9aa9b4417baf0bb0aa325f82961dcb40.91ccf12a72965a38c5c6.js?__WB_REVISION__=9cc6ef1d68b0d2f035fff35298a3ed94:1 Error while reading CSS rules from https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.48.2/theme/zenburn.min.css SecurityError: Failed to read the 'cssRules' property from 'CSSStyleSheet': Cannot access rules
```

**Comment by mfix22** [comment]
@cyberpirate92 I'm going to look into this tomorrow. I assume this is just an encoding race-condition from the `%c`

**Comment by mfix22** [comment]
@cyberpirate92 yup, this was just what I expected (previous issues have depicted the same issue). We have to encode the string in order for `dom-to-image` to create the dataUrl correctly, but that encoding temporarily breaks the syntax highlighting. 

These are the lines that do the temporary encoding: https://github.com/carbon-app/carbon/blob/main/components/Editor.js#L136-L142

I am not sure what to do about this in the moment, but will leave this open in case others do.

**Comment by mfix22** [comment]
Closed by #1069
